### PR TITLE
Implement Package#add_file view in bootstrap

### PR DESCRIPTION
**Pull request by ChrisBr**
image

vs. 

image

### CODE REVIEW

**Comment by Ana06** [comment]
The [review app](http://obs-reviewlab.opensuse.org/chrisbr-bootstrapfile-upload) is broken... :sob:

**Comment by ChrisBr** [comment]
> The review app is broken... sob

You're too fast :smile: Give it some time ... The review app usually takes up to 5 minutes to start, especially as we have now many review apps at the same time running. So if you get a 503, just come back in a few minutes ...

File: src/api/app/views/webui2/webui/package/add_file.html.haml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+- @pagetitle = 'Add File'
+
+.card.package-add-file
+  = render partial: 'tabs', locals: { project: @project, package: @package }
+  .card-body
+    %h3
+      Add file to #{@project.name}/#{@package.name}
```
Comment (dmarcoux) [inline src/api/app/views/webui2/webui/package/add_file.html.haml]: I believe this should be the same as `@pagetitle`

Comment (ChrisBr) [reply src/api/app/views/webui2/webui/package/add_file.html.haml]: Should it?
I was not aware of that :cry: Do we have this documented somewhere?

Comment (dmarcoux) [reply src/api/app/views/webui2/webui/package/add_file.html.haml]: I will document it in the Bootstrap wiki page.

Comment (dmarcoux) [reply src/api/app/views/webui2/webui/package/add_file.html.haml]: Done here: https://github.com/openSUSE/open-build-service/wiki/How-to-work-in-Bootstrap-(new-UI)#implementation-details

**Comment by bgeuken** [comment]
@ChrisBr Also, when creating a file from a remote URL I end up with this (though this also happens with the old UI :sad:):

```xml
<services>
  <service name="download_url">
    <param name="host">github.com</param>
    <param name="protocol">https</param>
    <param name="path">/openSUSE/open-build-service/blob/master/CONTRIBUTING.md</param>
    <param name="filename">foo</param>
  </service>
</services>
```
Form input was:
Filename: foo
Remote URL: https://github.com/openSUSE/open-build-service/blob/master/CONTRIBUTING.md

**Comment by ChrisBr** [comment]
> Adding git URLs isn't possible because of the client side validation, for example git@github.com:openSUSE/open-build-service.git.

Right because there is a HTML5 URL validator on the field! I will remove it.

**Comment by bgeuken** [comment]
I actually like the validation. Maybe we can just loosen it a bit, like ` <input type="url" id="url" pattern="https://.*" required /> ` (from [https://developer.mozilla.org](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/url))

**Comment by ChrisBr** [comment]
> Adding git URLs isn't possible because of the client side validation, for example git@github.com:openSUSE/open-build-service.git.

Double checked with build.o.o and this URL is also not supported there. However, this URL is supported ``git://github.com/openSUSE/open-build-service.git``, so you need to have the protocol at the beginning. This is also valid with the HTML 5 input URL validator because it only validates ``://`` presence it seems.

So I would say it is fine for now?

**Comment by ChrisBr** [comment]
> When uploading a file from my desktop I get:
" Error while creating '' file: undefined method `original_filename' for "Screenshot from 2018-09-25 13-13-33.png":String."

Thanks for catching this :cry: I really uploaded many files but then Rubocop messed up the ``multipart: true`` in the form header, this was causing this. Anyway, thanks for reviewing in detail and catching it :heart:

File: src/api/app/views/webui2/webui/package/add_file.html.haml (hunk 1 above)
Comment (ChrisBr) [reply src/api/app/views/webui2/webui/package/add_file.html.haml]: Updated!

**Pull request merged**